app/packages/utilities/src/type-check.ts

```diff
@@ -18,6 +18,23 @@ export function isObjectIdString(value: string, strict = true) {
   return isHex(value) && strict ? value.length === 24 : value.length <= 24;
 }
 
+export function isFunctionalComponent(
```

Comment:
This seems out of the ordinary...

1. it is named in a way that makes me think only things like `const MyComp = () => <h1>Hello</h1>` would return `true` ...I wouldn't expect `useEffect` to return `true`. We should clarify this somehow.
2. Its odd that we need this... it might be self explanatory but adding an explanation for why it is useful / needed would help clarify this for our future selves.